src/Metrics.ts

```diff
@@ -54,7 +72,28 @@ const ReadWrite = {
     Represent a single DynamoDB table
  */
 export class Metrics {
-    constructor(table, params = {}, prior = {}) {
+    log: Log | SenseLogs
+    metrics: OneMetricsParams
+    table: Table
+
+    count: number
+    counters: {
+        totals?: { count: number, latency: number, read: number, requests: number, scanned: number, write: number},
+        dimensions?: string[],
+        dimensionValues?: {
+            Table: string,
+            Tenant: string,
+            Source: string,
+            Index: string,
+            Model: string,
+            Operation: string,
+            Profile: string
+        },
+        properties?: Record<string, string>,
+    }
```

Comment:
I inferred this from reading the code but not 100% I got this right.

Reply:
Probably should have a defined type rather than inline that. addMetric has a similar inline declaration.

Looks reasonable for now.

Reply:
Yea. I think there's going to be a long tail of clean up. First thing is just to get TS building happily and tests passing. Then deal with that.